### PR TITLE
Fix linter errors in migrations

### PR DESCRIPTION
**Pull request by brymut**
Refs #1774 Fix linter errors in migrations except for recently created missing-backwards-migration-callable. Added `--module-naming-style=any` to silence invalid-name errors in migration module names.

### CODE REVIEW

**Comment by brymut** [comment]
For the module name not following the snake_case related errors, we could possibly use the [predefined naming styles option](http://pylint.pycqa.org/en/latest/user_guide/options.html?highlight=Naming%20styles#predefined-naming-styles) 
`--module-naming-style=any` to silence them.

**Comment by atodorov** [comment]
> For the module name not following snake_case related errors, we could possibly use the [predefined naming styles option](http://pylint.pycqa.org/en/latest/user_guide/options.html?highlight=Naming%20styles#predefined-naming-styles)
> `--module-naming-style=any` to silence them. @atodorov

Is this a cli option, I've never used it ? Sounds good to me though.

File: tcms/testcases/migrations/0006_merge_text_field_into_testcase_model.py [hunk 1]
```diff
@@ -20,17 +20,17 @@ def convert_test_case_text(test_case_text):
...
-    TestCase = apps.get_model('testcases', 'TestCase')
-    TestCaseText = apps.get_model('testcases', 'TestCaseText')
-    HistoricalTestCase = apps.get_model('testcases', 'HistoricalTestCase')
+    test_case = apps.get_model('testcases', 'TestCase')
+    test_case_text = apps.get_model('testcases', 'TestCaseText')
+    historical_test_case = apps.get_model('testcases', 'HistoricalTestCase')
 
-    for test_case in TestCase.objects.all():
-        latest_text = TestCaseText.objects.filter(case=test_case.pk).order_by('-pk').first()
+    for test_case in test_case.objects.all():
```
Comment (atodorov) [inline tcms/testcases/migrations/0006_merge_text_field_into_testcase_model.py:27]: loop variable the same as the variable holding the model - could lead to hard to debug issues. Don't use underscores for the model names.

File: tcms/testcases/migrations/0007_convert_is_automated_to_boolean.py [hunk 2]
```diff
@@ -4,14 +4,14 @@
 
 
 def forward_migrate_data(apps, schema_editor):
-    TestCase = apps.get_model('testcases', 'TestCase')
-    HistoricalTestCase = apps.get_model('testcases', 'HistoricalTestCase')
+    test_case = apps.get_model('testcases', 'TestCase')
+    historical_test_case = apps.get_model('testcases', 'HistoricalTestCase')
 
-    for test_case in TestCase.objects.all():
+    for test_case in test_case.objects.all():
```
Comment (atodorov) [inline tcms/testcases/migrations/0007_convert_is_automated_to_boolean.py:10]: Same comment as above about loop variable name vs. model name.

File: tcms/testcases/migrations/0009_populate_missing_text_history.py [hunk 3]
```diff
@@ -2,11 +2,11 @@
...
 
 def forward_copy_data(apps, schema_editor):
-    TestCase = apps.get_model('testcases', 'TestCase')
-    HistoricalTestCase = apps.get_model('testcases', 'HistoricalTestCase')
+    test_case = apps.get_model('testcases', 'TestCase')
+    historical_test_case = apps.get_model('testcases', 'HistoricalTestCase')
 
-    for test_case in TestCase.objects.all():
-        history = HistoricalTestCase.objects.filter(
+    for test_case in test_case.objects.all():
```
Comment (atodorov) [inline tcms/testcases/migrations/0009_populate_missing_text_history.py:8]: Same comment as above about loop variable name vs. model name.

File: tcms/testcases/migrations/0010_remove_bug.py [hunk 4]
```diff
@@ -2,14 +2,14 @@
 
 
 def forward_copy_data(apps, schema_editor):
-    Bug = apps.get_model('testcases', 'Bug')
-    LinkReference = apps.get_model('linkreference', 'LinkReference')
+    bug = apps.get_model('testcases', 'Bug')
+    link_reference = apps.get_model('linkreference', 'LinkReference')
 
-    for bug in Bug.objects.all():
+    for bug in bug.objects.all():
```
Comment (atodorov) [inline tcms/testcases/migrations/0010_remove_bug.py:8]: Same comment as above about loop variable name vs. model name. Use `bugmodel` or something like that.

File: tcms/testruns/migrations/0006_rename_test_case_run_to_test_execution.py [hunk 5]
```diff
@@ -5,9 +5,9 @@
 
 
 def rename_permissions(apps, schema_editor):
-    Permission = apps.get_model('auth', 'Permission')
+    permission = apps.get_model('auth', 'Permission')
 
-    for permission in Permission.objects.filter(codename__contains='testcaserun'):
+    for permission in permission.objects.filter(codename__contains='testcaserun'):
```
Comment (atodorov) [inline tcms/testruns/migrations/0006_rename_test_case_run_to_test_execution.py:10]: Same comment as above about loop variable name vs. model name. Use "permission_model" or something like that.

File: tcms/testcases/migrations/0009_populate_missing_text_history.py [hunk 6]
```diff
@@ -7,8 +7,8 @@ def forward_copy_data(apps, schema_editor):
 
     for test_case in test_case.objects.all():
         history = historical_test_case.objects.filter(
-                    case_id=test_case.pk
-                  ).order_by('-history_id').first()
+            case_id=test_case.pk
+            ).order_by('-history_id').first()
```
Comment (atodorov) [inline tcms/testcases/migrations/0009_populate_missing_text_history.py:11]: I think line 11 should be 4 spaces to the left, on the same level as the history variable above but I don't see pylint complaining. You could also move everything to the line above.

**Comment by atodorov** [comment]
From the pylint logs I also see:

************* Module tcms.testruns.migrations.0006_rename_test_case_run_to_test_execution
tcms/testruns/migrations/0006_rename_test_case_run_to_test_execution.py:1:0: C0103: Module name "0006_rename_test_case_run_to_test_execution" doesn't conform to snake_case naming style (invalid-name)
tcms/testruns/migrations/0006_rename_test_case_run_to_test_execution.py:52:8: W5197: tcms.testruns.migrations.0006_rename_test_case_run_to_test_execution Always include backwards migration callable (missing-backwards-migration-callable)

Notice how the module name is repeated after W5197 and then comes the actual error message. I think this is because of the `args=module.name` parameter in the snippet below:

```
        if node.func.as_string().endswith('RunPython') and len(node.args) < 2:
            if node.keywords:
                for keyword in node.keywords:
                    if keyword.arg == 'reverse_code':
                        return
                self.add_message('missing-backwards-migration-callable',
                                 args=module.name, node=node)
            else:
                self.add_message('missing-backwards-migration-callable',
                                 args=module.name, node=node)
```


^^^ that's from your linter in pylint-django. Would you mind sending a patch for it too.

File: tcms/testcases/migrations/0009_populate_missing_text_history.py (hunk 6 above)
Comment (brymut) [reply tcms/testcases/migrations/0009_populate_missing_text_history.py:11]: I think moving it all to line 10 works best here.

File: tcms/testcases/migrations/0001_initial.py [hunk 7]
```diff
@@ -281,10 +281,10 @@ class Migration(migrations.Migration):
...
                                                   verbose_name='Type')),
                 ('base_url', models.CharField(
-                        max_length=1024, null=True, blank=True,
-                        verbose_name='Base URL',
-                        help_text="""Base URL, for example <strong>https://bugzilla.example.com</strong>!
-Leave empty to disable!
+                    max_length=1024, null=True, blank=True,
+                    verbose_name='Base URL',
+                    help_text="""Base URL, for example
+ <strong>https://bugzilla.example.com</strong>! Leave empty to disable!
```
Comment (atodorov) [inline tcms/testcases/migrations/0001_initial.py:287]: This change is causing the test to fail b/c in the model the newlines in the string are in a different place.

**Comment by brymut** [comment]
I think its already on latest

**Pull request merged**